Makefile.js

```diff
@@ -343,19 +359,33 @@ function generatePrerelease(prereleaseId) {
 function publishRelease() {
     ReleaseOps.publishRelease();
     const releaseInfo = JSON.parse(cat(".eslint-release-info.json"));
-    const isPreRelease = /[a-z]/u.test(releaseInfo.version);
 
-    /*
-     * for a pre-release, push to the "next" branch to trigger docs deploy
-     * for a release, push to the "latest" branch to trigger docs deploy
-     */
-    if (isPreRelease) {
-        exec("git push origin HEAD:next -f");
-    } else {
-        exec("git push origin HEAD:latest -f");
-    }
+    const docsSiteBranch = releaseInfo.packageTag === "maintenance"
+        ? `v${semver.major(releaseInfo.version)}.x`
+        : releaseInfo.packageTag; // "latest" or "next"
+
+    echo(`Updating docs site branch: ${docsSiteBranch}`);
+    exec(`git push origin HEAD:${docsSiteBranch} -f`);
 
     publishSite();
+
+    // Update changelog and list of versions on the main branch
+    if (getCurrentGitBranch() !== MAIN_GIT_BRANCH) {
+        echo(`Updating changelog and versions on branch: ${MAIN_GIT_BRANCH}`);
+
+        exec(`git checkout ${MAIN_GIT_BRANCH} --force`);
+
+        fs.writeFileSync(CHANGELOG_FILE, `${releaseInfo.markdownChangelog}${cat(CHANGELOG_FILE)}`);
```

Comment:
This will insert the changelog updates for the new maintenance release on top of the changelog on the main branch, right? Will users understand to which previous release those changes were applied upon, or would it make sense to indicate the branch name along with the version number (e.g. v8.57.1 on v8.x-dev)? I am looking at the [changelog updates for v8.57.0](https://github.com/eslint/eslint/blob/v9.10.0/CHANGELOG.md?plain=1#L533-L547) that was published sometime during the v9 pre-release phase, and now it's not obvious at what point the v8 and v9 branches diverged.

Reply:
> This will insert the changelog updates for the new maintenance release on top of the changelog on the main branch, right?

Yes, the idea was to order the releases chronologically, not according to semver rules.

> Will users understand to which previous release those changes were applied upon, or would it make sense to indicate the branch name along with the version number (e.g. v8.57.1 on v8.x-dev)? I am looking at the [changelog updates for v8.57.0](https://github.com/eslint/eslint/blob/v9.10.0/CHANGELOG.md?plain=1#L533-L547) that was published sometime during the v9 pre-release phase, and now it's not obvious at what point the v8 and v9 branches diverged.

I think this can be easily figured out from the commit history and tags, but open for other solutions. I believe it isn't a common practice to mention branches as they could be temporary and the process could change over time, but maybe we could add a link to the tag: https://github.com/eslint/eslint/tree/v8.57.0.

@nzakas what do you think?

Reply:
I think it's fine as-is. Realistically, I don't know how many people are going be looking at `CHANGELOG.md` on its own and trying to figure out releases from there. I think it's more likely that people will either view it via a commit that just shows the new entries or will look at GitHub releases or blog posts.

All that is to say: I don't think it's necessary to mention a tag or branch. People who really care will be able to figure out where to go to get that information.